spyglass/server.py

```diff
@@ -24,10 +24,10 @@ class StreamingServer(socketserver.ThreadingMixIn, server.HTTPServer):
     daemon_threads = True
 
 
-def run_server(bind_address, port, output):
-    class StreamingHandler(server.BaseHTTPRequestHandler):
+def run_server(bind_address, port, output, stream_url='/stream', snapshot_url='/snapshot'):
+    class StreamingHandler(server.BaseHTTPRequestHandler):        
         def do_GET(self):
-            if self.path == '/stream.mjpg':
+            if self.path in [stream_url, snapshot_url]:
```

Comment:
The Content-Type header states that the response contains multiple parts. This is due to the fact that the mjpeg stream is an unlimited series of images (in contrast to a video stream like h264). Technically the 'snapshot' can be seen as an mjpeg stream with only one frame. However, I assume it would be better to handle stream/snapshot separately and send the header `Content-Type: image/jpeg` without boundary.

This would be the same behavior as [ustreamer](https://github.com/pikvm/ustreamer/blob/d154a3d1c191608ed32d44724b45dab31fa6405a/src/ustreamer/http/server.c#L531)

Reply:
Didn't see that tbh.
Do you think a simple if else would be enough, as everything should be the same and would avoid code duplication.
If I'm wrong please correct me, I don't have that much knowledge about Internet Protocols.

Reply:
Sometimes it's ok to duplicate code. I've seen projects where avoiding code duplication lead to very hard to understand code.

Let me add add a commit as a suggestion.